mocha-appveyor-reporter.js

```diff
@@ -0,0 +1,131 @@
+var util = require('util'),
+    Base = require('mocha').reporters.Base;
+
+function AppVeyorReporter(runner) {
+  Base.call(this, runner);
+  var request = require('request-json'),
+    deasync = require('deasync'),
+    endpoint = process.env.APPVEYOR_API_URL,
+    log = console.log.bind(console),
+    error = console.error.bind(console),
+    warn = console.warn.bind(console),
+    logEntries = [],
+    client = endpoint ? request.createClient(endpoint) : null;
```

Comment:
`client = endpoint && request.createClient(endpoint)` is a bit more simplier IMO